piwheels/master/the_scribe.py

```diff
@@ -349,6 +361,76 @@ def write_project_page(self, package):
         except FileExistsError:
             pass
 
+    def delete_package(self, package):
+        """
+        Attempts to remove the index and project page directories (including all
+        known wheel files) of the specified *package*.
+
+        :param str package:
+            The name of the package to delete.
+        """
+        if len(package) == 0:
+            # refuse to delete /simple/ and /project/ by accident
+            self.logger.error('not deleting - package name is empty')
+            return
+
+        files = {f for f in self.db.get_package_files(package)}
+
+        for name in ('simple', 'project'):
+            self.logger.info('deleting /%s/ dir for %s', name, package)
+            pkg_dir = self.output_path / 'simple' / package
+            canon_dir = pkg_dir.with_name(canonicalize_name(pkg_dir.name))
+
+            for dir in (pkg_dir, canon_dir):
+                self.logger.debug('Processing %s', dir)
+                index = dir / 'index.html'
+                for file in files:
+                    file_path = dir / file
+                    try:
+                        file_path.unlink()
+                        self.logger.debug('File deleted: %s/%s', dir, file)
+                    except FileNotFoundError:
+                        self.logger.debug('File not found: %s/%s', dir, file)
+
+                try:
+                    index.unlink()
```

Comment:
Index deletion should occur *before* file deletion. When we write new packages, we ensure that the wheel is written first *then* the index to ensure that pip won't querying the index and go looking for a wheel that doesn't exist yet. Likewise, when removing packages we should reverse this and remove the index first, then the wheels.

Reply:
I did consider that but thought that the removal of the index file would leave the system open to the directory listing, which doubles-up as a simple index. Alternatively we could write out a blank index file first?

Reply:
Have we got Apache configured to provide dir-listings in the absence of index.html? If so we ought to disable that...